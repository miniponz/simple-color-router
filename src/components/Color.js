import React from 'react';
import PropTypes from 'prop-types';

function Color({ match }){

  return (
    <div style={ { width: '100px', height: '100px', backgroundColor: decodeURIComponent(match.params.color) } }/>
  );
}

Color.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      color: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

export default Color;
