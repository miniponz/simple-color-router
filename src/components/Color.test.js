import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Colorcomp', () => {
  it(' renders Color', () => {
    const match = { params: { color : 'blue' } }; 
    const wrapper = shallow(<Color match={match} />);
    expect(wrapper).toMatchSnapshot();
  });
});
