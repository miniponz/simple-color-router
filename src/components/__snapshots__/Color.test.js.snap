// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorcomp  renders Color 1`] = `
<div
  style={
    Object {
      "backgroundColor": "blue",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
